src/main/java/de/sakpaas/backend/util/RequestUtils.java

```diff
@@ -0,0 +1,52 @@
+package de.sakpaas.backend.util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import javax.servlet.http.HttpServletRequest;
+import lombok.SneakyThrows;
+
+/**
+ * An utils class for {@link HttpServletRequest} related methods. Using the Singleton approach for
+ * better mocking while testing.
+ */
+public final class RequestUtils {
+
+  private static final RequestUtils INSTANCE = new RequestUtils();
+
+  // Hide constructor
+  private RequestUtils() {}
+
+  /**
+   * Returns the connection hash for the given {@link HttpServletRequest}.
+   *
+   * @param request the {@link HttpServletRequest} to hash
+   * @return the SHA-256 hash
+   */
+  @SneakyThrows
+  public byte[] generateConnectionHash(HttpServletRequest request) {
+    MessageDigest digest = MessageDigest.getInstance("SHA-256");
+
+    // If the backend runs behind a reverse proxy, use the header
+    String remoteAddress = request.getHeader("X-FORWARDED-FOR");
+    // else use the connecting Address
+    if (remoteAddress == null || "".equals(remoteAddress)) {
```

Comment:
Warum nicht `remoteAddress.equals("")`?

Reply:
Ist in diesem Fall egal, da vorne ja schon gecheckt wird ob `remoteAddress` nicht `null` ist. Ich habe es ehrlicherweise einfach aus dem Internet kopiert. Einen Unterschied macht es aber nicht.